src/esp/gfx/GenericDrawable.cpp

```diff
@@ -183,6 +189,40 @@ void GenericDrawable::draw(const Mn::Matrix4& transformationMatrix,
     shader_->bindObjectIdTexture(*(materialData_->objectIdTexture));
   }
 
+  if (skinData_) {
+    // Gather joint transformations
+    const auto& skin = skinData_->skinData->skin;
+    const auto& transformNodes = skinData_->jointIdToTransformNode;
+    auto& jointIdToArticulatedObjectNodes =
+        skinData_->jointIdToArticulatedObjectNode;
+
+    ESP_CHECK(jointTransformations_.size() == skin->joints().size(),
+              "Joint transformation count doesn't match bone count.");
+
+    // Undo root node transform so that the model origin matches the root
+    // articulated object link.
+    const auto invRootTransform =
+        jointIdToArticulatedObjectNodes[skinData_->rootJointId]
+            ->absoluteTransformationMatrix()
+            .inverted();
+
+    auto lastTransform = Mn::Matrix4{Magnum::Math::IdentityInit};
+    for (std::size_t i = 0; i != jointTransformations_.size(); ++i) {
+      const auto jointNodeIt = transformNodes.find(skin->joints()[i]);
+      if (jointNodeIt != transformNodes.end()) {
+        jointTransformations_[i] =
+            invRootTransform *
+            jointNodeIt->second->absoluteTransformationMatrix() *
+            skin->inverseBindMatrices()[i];
+        lastTransform = jointTransformations_[i];
+      } else {
+        // Joint not found - use last transform.
+        jointTransformations_[i] = lastTransform;
```

Comment:
Ok, I understand now why this find failure isn't an error. As you say, some of the bones in the skinned model don't need to be driven from an articulated link. But why is `lastTransform` the correct thing to set it to?